emitterc.go

```diff
@@ -1847,7 +1847,7 @@ func yaml_emitter_write_double_quoted_scalar(emitter *yaml_emitter_t, value []by
 
 func yaml_emitter_write_block_scalar_hints(emitter *yaml_emitter_t, value []byte) bool {
 	if is_space(value, 0) || is_break(value, 0) {
-		indent_hint := []byte{'0' + byte(emitter.best_indent)}
+		indent_hint := []byte{'0' + byte(emitter.indent)}
```

Comment:
Thanks for the fix, it seems like this change does not work in some cases. For example

```
package main

import (
	"fmt"

	"gopkg.in/yaml.v3"
)

type T struct {
	F      int `yaml:"a,omitempty"`
	B      string
	Values []T
}

func main() {
	data := &T{B: "test1"}

	withNewLine := `
line 1
  line 2

  line 3

`
	data.Values = append(data.Values, T{B: withNewLine})
	val, _ := yaml.Marshal(data)
	fmt.Println(string(val))
}
```
produces 
```
b: test1
values:
    - b: |8+
        line 1
          line 2

          line 3

      values: []
```
which is invalid.

I found that the changes:
https://github.com/go-yaml/yaml/blob/6367991e90d61590c8621de8aa4e562966d0822a/emitterc.go#L727-L730
https://github.com/go-yaml/yaml/blob/6367991e90d61590c8621de8aa4e562966d0822a/emitterc.go#L237-L239

from the PR https://github.com/go-yaml/yaml/pull/640 for the issue that you linked worked.
Do you mind adjusting your PR with these changes? Thank you.

Reply:
Hi, sure

Reply:
@emanlodovice I applied those changes and added more test cases